GDJS/Runtime/runtimeobject.ts

```diff
@@ -2323,26 +2332,32 @@ namespace gdjs {
       }
 
       // Do a real check if necessary.
-      let testSqDist = closest ? raySqBoundingRadius : 0;
-      const hitBoxes = this.getHitBoxesAround(x, y, endX, endY);
-      for (const hitBox of hitBoxes) {
-        const res = gdjs.Polygon.raycastTest(hitBox, x, y, endX, endY);
-        if (res.collision) {
-          if (closest && res.closeSqDist < testSqDist) {
-            testSqDist = res.closeSqDist;
-            result = res;
-          } else {
-            if (
-              !closest &&
-              res.farSqDist > testSqDist &&
-              res.farSqDist <= raySqBoundingRadius
-            ) {
-              testSqDist = res.farSqDist;
-              result = res;
-            }
+      if (closest) {
+        let sqDistMin = raySqBoundingRadius;
+        const hitBoxes = this.getHitBoxesAround(x, y, endX, endY);
+        for (const hitBox of hitBoxes) {
+          const res = gdjs.Polygon.raycastTest(hitBox, x, y, endX, endY);
+          if (res.collision && res.closeSqDist < sqDistMin) {
```

Comment:
This condition will not allow `res.closeSqDist === raySqBoundingRadius`.
`let sqDistMin = Number.MAX_VALUE;` will also allow values that would be a bit more than raySqBoundingRadius because of eventual rounding error which make the code stronger because it's probably better to find something a bit too far that not finding something that is 1e-16 too far.

Reply:
Good catch, this was an issue. I used `Number.MAX_VALUE `.